File: Website_Phim/WebPhim-Backend/MovieWebsite/MovieWebsite/src/main/java/com/project/MovieWebsite/dtos/RateDTO.java
package com.project.MovieWebsite.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import lombok.*;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class RateDTO {

    @JsonProperty("movie_id")
    private int movieId;

    @JsonProperty("user_id")
    private int userId;

    @JsonProperty("number_stars")
    @Min(value = 1, message = "Number star must be >=1")
    @Max(value = 5, message = "Number star must be <=5")
    private int numStars;

}

File: MovieWebsite-fontend/src/app/dtos/user/rate.dto.ts
import{
    IsNumber
 }from 'class-validator'
 
 export class RateDTO{
     
    @IsNumber()
     movie_id: number;

    @IsNumber()
     user_id: number;

    @IsNumber()
    number_stars: number;
        
     constructor(data: any){
        this.movie_id= data.movie_id 
        this.user_id= data.user_id
        this.number_stars= data.number_stars
     }
 }